osquery/core/status.cpp

```diff
@@ -0,0 +1,26 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under both the Apache 2.0 license (found in the
+ *  LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ *  in the COPYING file in the root directory of this source tree).
+ *  You may select, at your option, one of the above-listed licenses.
+ */
+
+#include <osquery/logger.h>
+#include <osquery/status.h>
+
+namespace osquery {
+
+constexpr int Status::success_code;
+
+Status Status::failure(int code, std::string message) {
+  if (code == Status::success_code) {
+    LOG(ERROR) << "Using Status::failure to create Status object with "
```

Comment:
A good reminder that we could do a lot better with a culture of assets and CHECKs and running tests in dbg.

Reply:
I see possible danger from this method, because some errors may be handled and should not be reported. 1 handled but common failure may consume a lot of traffic and destroy logging system.

Reply:
Why not to add debug assert here, IMHO it should crash in debug builds, since it's clear logic error.